CppSamples/Maps/CreateDynamicBasemapGallery/BasemapStyleListModel.cpp

```diff
@@ -51,16 +51,21 @@ QVariant BasemapStyleListModel::data(const QModelIndex& index, int role) const
 
 int BasemapStyleListModel::rowCount(const QModelIndex& parent) const
 {
+    Q_UNUSED(parent)
     return m_previews.size();
 }
 
 void BasemapStyleListModel::insertItemsIntoGallery(const QList<BasemapStyleInfo*> infos)
 {
     beginResetModel();
     m_previews.clear();
-    for (const BasemapStyleInfo* info : infos)
+    for (auto index = 0; index < infos.size(); ++index)
     {
-        m_previews.append(std::move(info));
+        m_previews.insert(index, std::move(infos.at(index)));
```

Comment:
```suggestion
        m_previews.insert(index, infos.at(index));
```
Guillaume probably knows better than I do, but I assume infos.at is already an r-value so no need to std::move it.

Reply:
Looks like QList::at() is not rvalue, but still `std::move` is not necessary because of Qt's implicit sharing or copy-on-write mechanism.

Reply:
> infos.at is already an r-value

It's tricky. An function call is a lvalue if the return type is a lvalue-reference (including const) and a prvalue otherwise. https://en.cppreference.com/w/cpp/language/value_category

`QList::at` returns a const ref (https://doc.qt.io/qt-6/qlist.html#at) so it's a lvalue. Using `std::move` change nothing. 

But there is no copy, because `QList::at` returns a const ref and `QList::insert` takes a const ref.

TL;DR: `const` can be removed

Reply:
Thanks, removed `const`.